docs/src/_reference/command-line-interface.md

```diff
@@ -732,11 +731,69 @@ meltano job remove <job_name>
 
 ### Examples
 
+##### Tasks and Blocks
+
+A task in a job consists of one or more _blocks_. A block is simply a set of related plugins that will be executed serially.
+For example, you can create a job that consists of a single tap and target:
+
+```bash
+meltano job add tap-gitlab-to-target-postgres --tasks "tap-gitlab target-postgres"
+```
+
+This would add the following to your `meltano.yml`:
+
+```yaml
+jobs:
+  - name: tap-gitlab-to-target-postgres
+    tasks:
+      - tap-gitlab target-postgres
+```
+
+Plugins in the same block should be separated by a space and blocks should be separted by a comma.
+
+You can also provide tasks in YAML format. For example, the following is equivalent to the above:
+
+```bash
+meltano job add tap-gitlab-to-target-postgres --tasks "[tap-gitlab target-postgres]"
+```
+
+Note that any single block which includes a tap _must_ also include a target and vice versa.
+The tap must also precede the target.
+
+All of the following are invalid:
+
+```bash
+# A block with a tap must also include a target.
+meltano job add invalid-job --tasks "tap-gitlab"
+>>> Job 'invalid-job' has invalid tasks. block violates set requirements: Found no end in block set!
+
+# Within a block, a target must be preceded by a tap.
+meltano job add invalid-job --tasks target-postgres
+>>> Job 'invalid-job' has invalid tasks. block violates set requirements: Unknown command type or bad block sequence at index 1, starting block 'target-postgres'
+
+# Blocks are separated by commas, so Meltano considers this tap and target as two separate invalid blocks.
+meltano job add invalid-job --tasks "tap-gitlab, target-postgres"
```

Comment:
Would this have worked with e.g. `--tasks "dbt:seed, dbt run"`? I'd expect not, because I think the `[]` are missing that make this an array.

Reply:
`--tasks "dbt:seed, dbt:run"` would be accepted as a valid job but then would fail at runtime because you're right, it wouldn't be parsed as an array. meltano would try to invoke the `seed,` (note the trailing comma) command for `dbt` and so would fail.

Reply:
1. dbt:seed, dbt:run are independent commands. 
2. Thats still a single task, they'd be executed sequentially (assuming the first doesn't explode)

```yaml
# One task, two blocks
- name: demo
  tasks:
  - dbt:seed, dbt:run
```

```
(melty-3.8) ➜  rundev meltano run --dry-run demo
2022-06-09T00:31:28.773652Z [info     ] Setting 'console' handler log level to 'debug' for dry run
2022-06-09T00:31:28.855305Z [debug    ] expanding job to tasks         job_name=demo tasks=['dbt:seed,', 'dbt:run']
2022-06-09T00:31:29.584182Z [debug    ] Found plugin parent            parent=dbt plugin=dbt source=discovery
2022-06-09T00:31:29.592556Z [debug    ] plugin command added for execution command_name=seed, commands={0: 'seed,'} plugin_name=dbt
2022-06-09T00:31:29.592632Z [debug    ] found plugin in cli invocation plugin_name=dbt
2022-06-09T00:31:29.600597Z [debug    ] plugin command added for execution command_name=run commands={0: 'seed,', 1: 'run'} plugin_name=dbt
2022-06-09T00:31:29.600649Z [debug    ] found plugin in cli invocation plugin_name=dbt
2022-06-09T00:31:29.600718Z [debug    ] next block not extractor       offset=0 plugin_type=transformers
2022-06-09T00:31:29.600762Z [debug    ] found PluginCommand            offset=0 plugin_type=transformers
2022-06-09T00:31:29.651529Z [debug    ] next block not extractor       offset=1 plugin_type=transformers
2022-06-09T00:31:29.651617Z [debug    ] found PluginCommand            offset=1 plugin_type=transformers
2022-06-09T00:31:29.723809Z [debug    ] All ExtractLoadBlocks validated, starting execution.
2022-06-09T00:31:29.723977Z [info     ] Dry run, but would have run block 1/2. block_type=InvokerCommand comprised_of=dbt:seed,
2022-06-09T00:31:29.724066Z [info     ] Dry run, but would have run block 2/2. block_type=InvokerCommand comprised_of=dbt:run
```

Reply:
Ah i see what your saying. Yea `dbt:seed,` is being used (with a dangling comma). That comma is being treated as a string,  and we shouldn't be documenting it as its not valid  we can try to be proactive and throw an error at job creation, but its no different than this typo:

```
meltano job add demo2 --tasks="dbt:seed- dbt:run"
Added job demo2: ['dbt:seed- dbt:run']
```

Reply:
Yeah I'm realizing my example of `meltano job add invalid-job --tasks "tap-gitlab, target-postgres"` doesn't fail because it parses into a `tap-gitlab` followed by a `target-postgres` block and so fails because `tap-gitlab` isn't valid as a block on its own. It fails because it parses into `tap-gitlab,` (note the trailing comma) followed by a `target-postgres` block and so fails because `target-postgres` isn't a valid block on its own.